### ISSUE TITLE
Regression with Snowpack from 1.0.6 -> 1.0.7

### ISSUE THREAD

**Issue by water-a** [pmndrs/valtio#179]
# 🐛 bug report

## 🎛 Configuration (.babelrc, package.json, cli command)

index.html
```html
<html>
  <body>
    <script type="module" src="/index.js"></script>
  </body>
</html>
```

index.js
```js
import { proxy } from "valtio";
console.log(proxy);
```

package.json
```json
{
  "name": "reproduce",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "dependencies": {
    "react": "^17.0.2",
    "snowpack": "^3.8.2",
    "valtio": "^1.0.7"
  },
  "devDependencies": {},
  "scripts": {
    "start": "snowpack dev",
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}
```

Commands to run
```
npm i 
npm start
```

## 🤔 Expected Behavior
It should properly imports the `proxy` export from Valtio.

## 😯 Current Behavior
![image](https://user-images.githubusercontent.com/38114545/126111602-6e74412a-fd07-4567-9fdd-078673ab14f1.png)

## 🔦 Context
There seems to be a regression when using the latest valtio version and using it with Snowpack.

However if you downgrade to valtio@1.0.6, it properly imports the `proxy` export.
```
npm i valtio@1.0.6
npm start
```

**Comment by dai-shi** [comment]
Thanks for reporting. I was able to reproduce the issue with your instruction.

Looking at http://localhost:8080/_snowpack/pkg/valtio.v1.0.7.js,
it's using the esm/index.js, but from there it uses the cjs version of vanilla.js instead of esm/vanilla.js.

I'm not sure what is wrong. v1.0.6 was working because everything was cjs. As snowpack supports esm, we'd not want to use cjs. Hmmm..

cc @LXSMNSYC

**Comment by LXSMNSYC** [comment]
It seems that snowpack uses the Node's default behavior:

> Node.js treats JavaScript code as CommonJS modules by default. Authors can tell Node.js to treat JavaScript code as ECMAScript modules via the .mjs file extension, the package.json "type" field, or the --input-type flag. See Modules: Packages for more details.

so even if the package is sufficiently defined with exports, Node would still need the `"type"` field or the `mjs` file extension.

I tried this with node (`node index`) and `valtio/vanilla` still resolves to the CJS bundle (as proven by the bundle output from snowpack).

I'm not sure if specifying the "type" field may solve this issue (this requires both "proxy-compare" and "valtio" to resolve correctly.)

Note: I tested the same repro with Vite and it seems to resolve okay. It's either this is a Snowpack issue OR we fix "valtio" and "proxy-compare" to resolve correctly across bundlers (and Node).
